javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/ClassOrInterfaceDeclarationContextTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2007-2010 Júlio Vilmar Gesser.
+ * Copyright (C) 2011, 2013-2021 The JavaParser Team.
+ *
+ * This file is part of JavaParser.
+ *
+ * JavaParser can be used either under the terms of
+ * a) the GNU Lesser General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ * b) the terms of the Apache License
+ *
+ * You should have received a copy of both licenses in LICENCE.LGPL and
+ * LICENCE.APACHE. Please refer to those files for details.
+ *
+ * JavaParser is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ */
+
+package com.github.javaparser.symbolsolver.javaparsermodel.contexts;
+
+import com.github.javaparser.JavaParser;
+import com.github.javaparser.ParserConfiguration;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.resolution.types.ResolvedPrimitiveType;
+import com.github.javaparser.symbolsolver.JavaSymbolSolver;
+import com.github.javaparser.symbolsolver.javaparser.Navigator;
+import com.github.javaparser.symbolsolver.model.resolution.TypeSolver;
+import com.github.javaparser.symbolsolver.resolution.typesolvers.ReflectionTypeSolver;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class ClassOrInterfaceDeclarationContextTest {
+
+    private final TypeSolver typeSolver = new ReflectionTypeSolver();
+    private JavaParser javaParser;
+
+    @BeforeEach
+    void beforeEach() {
+        ParserConfiguration parserConfiguration = new ParserConfiguration();
+        parserConfiguration.setSymbolResolver(new JavaSymbolSolver(typeSolver));
+        javaParser = new JavaParser();
+    }
+
+    @Test
+    void testSolveWithoutTypeArguments() {
+        CompilationUnit alphaCU = parse("class Alpha { class Foo {} }");
+        ClassOrInterfaceDeclaration alpha = Navigator.demandClass(alphaCU, "Alpha");
+        ClassOrInterfaceDeclarationContext alphaContext = new ClassOrInterfaceDeclarationContext(alpha, typeSolver);
+
+        assertTrue(alphaContext.solveType("Foo").isSolved());
+        assertTrue(alphaContext.solveType("Foo", Collections.emptyList()).isSolved());
+        assertFalse(alphaContext.solveType("Foo", Collections.singletonList(ResolvedPrimitiveType.INT)).isSolved());
```

Comment:
Why is the type "Foo" resolved in this context?

Reply:
In this assertions, there are three things being confirmed:

`assertTrue(alphaContext.solveType("Foo").isSolved());`
We are making sure, that we are retro-compatible, so if someones is using the API without type parameters, the type is solved.

`assertTrue(alphaContext.solveType("Foo", Collections.emptyList()).isSolved());`
Looking at the class definition, the class Foo does not have type parameters. In this test, we make surethat trying to resolve it with a empty list it's resolved. Note that a empty list in type arguments is the same as saying it does not have type parameter.

`assertFalse(alphaContext.solveType("Foo", Collections.singletonList(ResolvedPrimitiveType.INT)).isSolved());`
Test to make sure, if we try to resolve Foo with a type parameter, it's not resolved since the declaration says it does not have one.